Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkAssetManager.cs

```diff
@@ -51,50 +54,80 @@ public static void OpenExampleScene()
+
+            var alcInstance = GetComponentsInScene<AudioLinkController>().FirstOrDefault();
+            if (alcInstance == null) AddPrefabInstance(_audioLinkControllerPath);
 #elif VRC_SDK_VRCSDK3 // VRC AVATAR
-                "Packages/com.llealloo.audiolink/Runtime/AudioLinkAvatar.prefab",
+            var alInstance = GetComponentsInScene<AudioLink>().FirstOrDefault();
+            audiolink = alInstance != null ? alInstance.gameObject : AddPrefabInstance(_audioLinkAvatarPath);
 #elif CVR_CCK_EXISTS // CVR
-                "Packages/com.llealloo.audiolink/Runtime/CVRAudioLink.prefab",
-                "Packages/com.llealloo.audiolink/Runtime/CVRAudioLinkController.prefab",
+            audiolink = GetGameObjectsInScene("CVRAudioLink").FirstOrDefault();
+            if (audiolink == null) audiolink = AddPrefabInstance(_audioLinkCVRPath);
+
+            var controller = GetGameObjectsInScene("CVRAudioLinkController").FirstOrDefault();
+            if (controller == null) AddPrefabInstance(_audioLinkCVRControllerPath);
 #else // Standalone
-                "Packages/com.llealloo.audiolink/Runtime/AudioLink.prefab",
+            var alInstance = GetComponentsInScene<AudioLink>().FirstOrDefault();
+            audiolink = alInstance != null ? alInstance.gameObject : AddPrefabInstance(_audioLinkPath);
 #endif
-            };
-            GameObject audiolink = null;
 
-            foreach (string path in paths)
+            if (audiolink != null)
             {
-                GameObject asset = AssetDatabase.LoadAssetAtPath<GameObject>(path);
-                if (asset != null)
-                {
-                    GameObject instance = (GameObject)PrefabUtility.InstantiatePrefab(asset);
-                    if (path.EndsWith("AudioLink.prefab"))
-                    {
-                        audiolink = instance;
-                    }
-                    EditorGUIUtility.PingObject(instance);
-                }
+                AudioLinkEditor.LinkAll(audiolink.GetComponent<AudioLink>());
+                EditorGUIUtility.PingObject(audiolink);
             }
+        }
 
-            if (audiolink != null)
+        private static GameObject AddPrefabInstance(string assetPath)
+        {
+            var sourceAsset = AssetDatabase.LoadAssetAtPath<GameObject>(assetPath);
+            GameObject instance = null;
+            if (sourceAsset != null)
             {
-                AudioLinkEditor.LinkAll(audiolink.GetComponent<AudioLink>());
+                instance = (GameObject)PrefabUtility.InstantiatePrefab(sourceAsset);
+                Undo.RegisterCreatedObjectUndo(instance, "Undo create prefab instance");
             }
+
+            return instance;
+        }
+
+        private static T[] GetComponentsInScene<T>(bool includeInactive = true) where T : Component
+        {
+            var stage = PrefabStageUtility.GetCurrentPrefabStage();
```

Comment:
can you TL;DR me why you need this?

Reply:
It's just a convenience method I pulled from my own scripts which finds the list of all components that exist within the current scene (or prefab preview stage if that's active). This way it can easily find AudioLink in the scene regardless of where the user parents it to without relying on gameobject names, which is a fragile search method.